core/core.go

```diff
@@ -303,6 +331,45 @@ func (c *Core) addToQueueIfNotAppended(block *types.Block) {
 	}
 }
 
+// SetSyncTarget sets the sync target entropy based on the prime blocks
+func (c *Core) SetSyncTarget(header *types.Header) {
+	if c.sl.subClients == nil || header.Hash() == c.sl.config.GenesisHash {
+		return
+	}
+
+	// Check if the header is in the badSyncTargets cache
+	_, ok := c.badSyncTargets.Get(header.Hash())
+	if ok {
+		return
+	}
+
+	nodeCtx := common.NodeLocation.Context()
+	// Set Sync Target for subs
+	if nodeCtx != common.ZONE_CTX {
+		if header != nil {
+			for i := range c.sl.subClients {
+				if c.sl.subClients[i] != nil {
+					c.sl.subClients[i].SetSyncTarget(context.Background(), header)
+				}
+			}
+		}
+	}
+	// In zone case
+	if c.syncTarget == nil || c.syncTarget.ParentEntropy().Cmp(header.ParentEntropy()) < 0 {
```

Comment:
Is there a reason we do this in all of the subs potentially getting a different result? Shouldn't we run this check in prime prior to sending to the subs?